elyra/templates/kfp_template.jinja2

```diff
@@ -8,6 +8,8 @@ from kfp_notebook.pipeline import NotebookOp
 )
 
 def create_pipeline():
+    {% set metrics_file = writable_container_dir ~ '/mlpipeline-metrics.json' %}
+    {% set metadata_file = writable_container_dir ~ '/mlpipeline-ui-metadata.json' %}
```

Comment:
I'm not very familiar with jinja intricacies but how is the value of `writable_container_dir` getting injected?  Does that need to be a parameter as well?

Reply:
the writable_container_dir is getting injected here as part of the template render: 
https://github.com/elyra-ai/elyra/blob/4ac7283da729c3241bc66854a6628433a9e39404/elyra/pipeline/processor_kfp.py#L173

Reply:
Right.  I didn't know whether we have a preference:

### assemble variable values in the template

```
            python_output = template.render(operations_list=defined_pipeline,
                                            pipeline_name=pipeline_name,
                                            api_endpoint=api_endpoint,
                                            pipeline_description=description,
                                            writable_container_dir=wcd)
```

vs 

### pass complete variable values 

```
            python_output = template.render(operations_list=defined_pipeline,
                                            pipeline_name=pipeline_name,
                                            api_endpoint=api_endpoint,
                                            pipeline_description=description,
                                            metrics_file='....json',
                                            metadata_file='....json')
```

Reply:
I would prefer the first way, is neater

Reply:
Ok - I also agree with the first approach. 

I thought anything specified in the `render()` call was required to have surrounding braces (e.g., `{{ writable_container_dir }}`) in the template, but there's more going on than that here that I'm not quite following.

Why isn't `{% set metrics_file = writable_container_dir ~ '/mlpipeline-metrics.json' %}` not
`{% set metrics_file = {{ writable_container_dir }} ~ '/mlpipeline-metrics.json' %}`?

Reply:
I thought so too initially but in [`{%  ... %}` statements](https://jinja.palletsprojects.com/en/2.11.x/templates/#list-of-control-structures) variables don't need to be written using expressions.

Reply:
I see. Thank you.